ProcessMaker/Events/AuthClientUpdated.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace ProcessMaker\Events;
+
+use Illuminate\Foundation\Events\Dispatchable;
+use ProcessMaker\Contracts\SecurityLogEventInterface;
+
+class AuthClientUpdated implements SecurityLogEventInterface
+{
+    use Dispatchable;
+
+    public $original_values;
+    public $changed_values;
+    public $data;
+    public $changes;
+
+    /**
+     * Create a new event instance.
+     *
+     * @return void
+     */
+    public function __construct(array $original_values, array $changed_values)
+    {
+        $this->original_values = $original_values;
+        $this->changed_values = $changed_values;
+        $this->buildData();
+        $this->changes = $changed_values;
+    }
+
+    /**
+     * Building the data
+     */
+    public function buildData() {
+        $this->data = [
```

Comment:
The FormatSecurityLogChanges trait does this job.

Also consider that ProcessMaker uses camelCase for variable and attribute names.

I can suggest the following changes.

```php
<?php

namespace ProcessMaker\Events;

use Illuminate\Foundation\Events\Dispatchable;
use ProcessMaker\Contracts\SecurityLogEventInterface;
use ProcessMaker\Traits\FormatSecurityLogChanges;

class AuthClientUpdated implements SecurityLogEventInterface
{
    use Dispatchable, FormatSecurityLogChanges;

    public $data;
    public $original;
    public $changes;
    private int $clientId;

    /**
     * Create a new event instance.
     *
     * @param int $clientId
     * @param array $original
     * @param array $changes
     */
    public function __construct($clientId, array $original, array $changes)
    {
        $this->clientId = $clientId;
        $this->original = $original;
        $this->changes = $changes;
        $this->buildData();
    }

    /**
     * Building the data
     */
    public function buildData() {
        $this->data = array_merge([
            'Auth Client Id' => $this->clientId
        ], $this->formatChanges($this->changes, $this->original));
    }
    
    /**
     * Return event data 
     */
    public function getData(): array
    {
        return $this->data;
    }
    
    /**
     * Return event changes 
     */
    public function getChanges(): array
    {
        return $this->changes;
    }

    /**
     * return event name
     */
    public function getEventName(): string
    {
        return 'AuthClientUpdated';
    }
}

```